src/shims/foreign_items.rs

```diff
@@ -276,7 +272,10 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                 )?;
             }
             "__rust_realloc" => {
-                let ptr = this.read_scalar(args[0])?.to_ptr()?;
+                let ptr = match this.read_scalar(args[0])?.not_undef()? {
```

Comment:
Uh, matching on a `Scalar` is extremely suspicious. For example, someone might cast a ptr to int and then pass that here, and that should work.

Reply:
oh damn, right I forgot to address this correctly. When I started I was like "check each place for whether it should be `force_ptr`", and then I forgot to do that.